activerecord/lib/active_record/persistence.rb

```diff
@@ -308,17 +308,23 @@ def instantiate(attributes, column_types = {}, &block)
       # for updating all records in a single query.
       def update(id = :all, attributes)
         if id.is_a?(Array)
+          if id.any? { |one_id| one_id.is_a?(ActiveRecord::Base) }
```

Comment:
```suggestion
          if id.any?(ActiveRecord::Base)
```

This means going back to using `===` internally, but I think @HParker's objection was to mixing `is_a?` and `===` on the same line, not to using `===` in general.